contracts/VestingEscrowFullyRevokable.vy

```diff
@@ -0,0 +1,290 @@
+    """
+    # ensure that the original contract cannot be initialized
+    self.initialized = True
+    LIDO_VOTING_CONTRACT_ADDR = voting_addr
+    SNAPSHOT_DELEGATE_CONTRACT_ADDR = snapshot_addr
+
+
+@external
+@nonreentrant('lock')
+def initialize(
+    admin: address,
+    token: address,
+    recipient: address,
+    amount: uint256,
+    start_time: uint256,
+    end_time: uint256,
+    cliff_length: uint256,
+) -> bool:
+    """
+    @notice Initialize the contract.
+    @dev This function is separate from `__init__` because of the factory pattern
+         used in `VestingEscrowFactory.deploy_vesting_contract`. It may be called
+         once per deployment.
+    @param admin Admin address
+    @param token Address of the ERC20 token being distributed
+    @param recipient Address to vest tokens for
+    @param amount Amount of tokens being vested for `recipient`
+    @param start_time Epoch time at which token distribution starts
+    @param end_time Time until everything should be vested
+    @param cliff_length Duration after which the first portion vests
+    """
+    assert not self.initialized, "can only initialize once"
+    self.initialized = True
+
+    self.token = ERC20(token)
+    self.admin = admin
+    self.start_time = start_time
+    self.end_time = end_time
+    self.cliff_length = cliff_length
+    self.is_fully_revoked = False
+
+    assert self.token.transferFrom(msg.sender, self, amount), "could not fund escrow"
+
+    self.recipient = recipient
+    self.disabled_at = end_time  # Set to maximum time
+    self.total_locked = amount
+    log Fund(recipient, amount)
+
+    return True
+
+
+@internal
+@view
+def _total_vested_at(time: uint256 = block.timestamp) -> uint256:
+    start: uint256 = self.start_time
+    end: uint256 = self.end_time
+    locked: uint256 = self.total_locked
+    if time < start + self.cliff_length:
+        return 0
+    return min(locked * (time - start) / (end - start), locked)
+
+
+@internal
+@view
+def _unclaimed(time: uint256 = block.timestamp) -> uint256:
+    if not self.is_fully_revoked:
+        return self._total_vested_at(time) - self.total_claimed
+    return 0
+
+
+@external
+@view
+def unclaimed() -> uint256:
+    """
+    @notice Get the number of unclaimed, vested tokens for recipient
+    """
+    # NOTE: if `revoke_unvested` is activated, limit by the activation timestamp
+    return self._unclaimed(min(block.timestamp, self.disabled_at))
+
+
+@internal
+@view
+def _locked(time: uint256 = block.timestamp) -> uint256:
+    return min(self.total_locked - self._total_vested_at(time), self.token.balanceOf(self))
+
+
+@external
+@view
+def locked() -> uint256:
+    """
+    @notice Get the number of locked tokens for recipient
+    """
+    # NOTE: if `rug_pull` is activated, limit by the activation timestamp
+    return self._locked(min(block.timestamp, self.disabled_at))
+
+
+@external
+def claim(beneficiary: address = msg.sender, amount: uint256 = MAX_UINT256):
```

Comment:
UserWarning: MAX_UINT256 is deprecated. Please use `max_value(uint256)` instead.

Reply:
Done